src/reference/asciidoc/mongodb.adoc

```diff
@@ -290,3 +290,117 @@ and other attributes that are common across all other inbound adapters (e.g., 'c
+* `mongo-template` - reference to an instance of `o.s.data.mongodb.core.MongoTemplate` (NOTE: you can not have both mongo-template and mongodb-factory set)
+* `entity-class` - The fully qualified name of the entity class to be passed to `find(..)` or `findOne(..)` method in MongoTemplate. If this attribute is not provided the default value is `org.bson.Document`.
+* `query` or `query-expression` - Specifies the MongoDb query. Please refer to http://www.mongodb.org/display/DOCS/Querying[MongoDB documentation] for more query samples.
+
+==== Configuring with Java Configuration
+
+The following Spring Boot application provides an example of configuring the outbound gateway using Java configuration:
+
+[source, java]
+----
+@SpringBootApplication
+public class MongoDbJavaApplication {
+
+    public static void main(String[] args) {
+        new SpringApplicationBuilder(MongoDbJavaApplication.class)
+            .web(false)
+            .run(args);
+    }
+
+    @Autowired
+    private MongoDbFactory mongoDbFactory;
+
+    @Bean
+    public MessageChannel requestChannel() {
+        return new DirectChannel();
+    }
+
+    @Bean
+    public MessageChannel replyChannel() {
+        return new QueueChannel(5);
+    }
+
+    @Bean
+    @ServiceActivator(inputChannel = "requestChannel")
+    public MessageHandler mongoDbOutboundGateway() {
+        MongoDbOutboundGateway gateway = new MongoDbOutboundGateway(this.mongoDbFactory);
+        gateway.setCollectionNameExpression(new LiteralExpression("foo"));
+        gateway.setMongoDbQuery("{'name' : 'Bob'}");
+        gateway.setExpectSingleResult(true);
+        gateway.setEntityClass(Person.class);
+        gateway.setOutputChannelName("replyChannel");
+
+        return gateway;
+    }
+
+    @Bean
+    @ServiceActivator(inputChannel = "replyChannel", poller =  @Poller(fixedDelay = "2000"))
+    public MessageHandler handler() {
+        return message -> System.out.println(message.getPayload());
+    }
+}
+----
+
+==== Configuring with the Java DSL
+
+The following Spring Boot application provides an example of configuring the Outbound Gateway using the Java DSL:
+
+[source, java]
+----
+@SpringBootApplication
+public class MongoDbJavaApplication {
+
+    public static void main(String[] args) {
+        new SpringApplicationBuilder(MongoDbJavaApplication.class)
+            .web(false)
+            .run(args);
+    }
+
+    @Autowired
+    private MongoDbFactory;
+
+    @Autowired
+    private MongoConverter
+
+
+    @Bean
+    public IntegrationFlow gatewaySingleQueryFlow() {
+        return f -> f
+                .handle(queryOutboundGateway())
+                .channel(c -> c.queue("retrieveResults"));
+    }
+
+    private MongoDbOutboundGatewaySpec queryOutboundGateway() {
+        return MongoDb.outboundGateway(this.mongoDbFactory, this.mongoConverter)
+                .query("{name : 'Bob'}")
+                .collectionNameExpression(new LiteralExpression("foo"))
```

Comment:
We talked before that Java DSL should be free from `Expression` API.
Therefore that `collectionNameExpression()` should be with the `String` instead.